lib/plugins/aws/package/compile/functions.js

```diff
@@ -43,6 +45,19 @@ class AwsCompileFunctions {
               '"awsKmsKeyArn" function property will be replaced by "kmsKeyArn"'
           );
         }
+        if (
+          !this.serverless.service.provider.lambdaHashingVersion &&
+          (this.serverless.service.provider.versionFunctions === true ||
+            Object.values(this.serverless.service.functions).some(
+              ({ versionFunction }) => versionFunction
+            ))
+        ) {
+          this.serverless._logDeprecation(
+            'LAMBDA_HASHING_VERSION_20201221',
```

Comment:
I think code can be _major_ specific and not hashing version specific, so maybe `LAMBDA_HASHING_VERSION_V2`. What do you think?

It's not a problem if assuming we again improve versioning we bump version code, and use same deprecation code in context of same major.
While in next major it'll be good to differentiate (as we gather stats about deprecations, and it relies on assumption that each code is _major_ specific)

Reply:
I wasn't sure about approach with this deprecation code and I think your idea makes sense here. I'll change it :+1: